packages/hyperion-flowlet/src/FlowletWrappers.ts

```diff
@@ -181,14 +181,56 @@ export function initFlowletTrackers(flowletManager: FlowletManager) {
     })
   }
 
-  IPromise.constructor.onValueObserverAdd(value => {
-    if (!getTriggerFlowlet(value)) {
-      const triggerFlowlet = flowletManager.top()?.data.triggerFlowlet;
-      if (triggerFlowlet) {
-        setTriggerFlowlet(value, triggerFlowlet);
+  for (const fi of [
+    IPromise.constructor,
+    IPromise.resolve,
+    IPromise.reject
+  ]) {
+    fi.onValueObserverAdd(value => {
+      if (!getTriggerFlowlet(value)) {
+        const triggerFlowlet = flowletManager.top()?.data.triggerFlowlet;
+        if (triggerFlowlet) {
+          setTriggerFlowlet(value, triggerFlowlet);
+        }
       }
-    }
-  });
+    });
+  }
+
+  for (const fi of [
+    IPromise.all,
+    IPromise.allSettled,
+    IPromise.any,
+    IPromise.race
+  ]) {
+    fi.onArgsAndValueMapperAdd(args => {
+      const iterable = args[0];
+      return value => {
+        const topTriggerFlowlet = getTriggerFlowlet(value) ?? flowletManager.top()?.data.triggerFlowlet;
+        const triggerFlowlets: TriggerFlowlet[] = [];
+        // Args can be iterable, so we use the modern for-loop https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all#parameters
+        for (const arg of iterable) {
+          if (arg instanceof Promise) {
+            const triggerFlowlet = getTriggerFlowlet(arg);
+            if (triggerFlowlet) {
+              triggerFlowlets.push(triggerFlowlet);
+            }
+          }
+        }
+
+        if (triggerFlowlets.length > 0) {
+          const flowletName = `${fi.name}{${triggerFlowlets.map(f => f.id).join(",")}}`;
```

Comment:
Maybe also append `Promise.` at the beginning- I think people will not know that it's related to Promise when they just see `all` for instance

Also wondering if we should use `()` instead of `{}`, since we've typically been using `()` to add extra info for other flowlets

Reply:
Good suggestion, updated.